### ISSUE TITLE
Improve findNewTarget in lookclose trait

### ISSUE THREAD

**Issue by elgbar** [CitizensDev/Citizens2#2442]
In `findNewTarget()` the closest player is found by sorting the array of players nearby. This could be improved to only find the closest player. A full sort of the array is unnecessary.

## Potential solution

Replace the following current code

```java
 if (!nearby.isEmpty()) {
    nearby.sort((o1, o2) -> {
        Location l1 = o1.getLocation(CACHE_LOCATION);
        Location l2 = o2.getLocation(CACHE_LOCATION2);
        if (!NPC_LOCATION.getWorld().equals(l1.getWorld()) || !NPC_LOCATION.getWorld().equals(l2.getWorld())) {
            return -1;
        }
        return Double.compare(l1.distanceSquared(NPC_LOCATION), l2.distanceSquared(NPC_LOCATION));
    });
    lookingAt = nearby.get(0);
}
```

with this code

```java
Optional<Player> closest = nearby.stream().min((o1, o2) -> {
    Location l1 = o1.getLocation(CACHE_LOCATION);
    Location l2 = o2.getLocation(CACHE_LOCATION2);
    if (!NPC_LOCATION.getWorld().equals(l1.getWorld()) || !NPC_LOCATION.getWorld().equals(l2.getWorld())) {
        return Integer.MAX_VALUE;
    }
    return Double.compare(l1.distanceSquared(NPC_LOCATION), l2.distanceSquared(NPC_LOCATION));
});
lookingAt = closest.orElse(null);
```

## Benefits

1. Fixes invalid player potentially being returned, if either of the players are in the wrong world
2. Improves runtime from `O(n log(n))` to `O(n)`

The output of command /version on my server is: This server is running Paper version git-Paper-436 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
The output of command /version citizens on my server is: Citizens version 2.0.27-SNAPSHOT (build 2226)

**Comment by elgbar** [comment]
An even faster way of find new look close target is as follows

```java
  public void findNewTarget() {
    Player closest = null;
    double minDistance = range * range;
    for (Entity entity : npc.getEntity().getNearbyEntities(range, range, range)) {
      if (entity instanceof Player) {
        Player player = (Player) entity;
        Location location = player.getLocation(CACHE_LOCATION);
        if (location.getWorld() != NPC_LOCATION.getWorld()) {
          continue;
        }

        double dist = location.distanceSquared(NPC_LOCATION);
        if (dist < minDistance
            && CitizensAPI.getNPCRegistry().getNPC(entity) == null
            && !isInvisible(player)) {
          closest = player;
          minDistance = dist;
        }
      }
    }

    if (closest != lookingAt) {
      new NPCLookCloseChangeTarget(npc, this, closest, lookingAt).callEvent();
      lookingAt = closest;
    }
  }
```

**Comment by elgbar** [comment]
It seems @fullwall have implemented my requested change with [fe3f12e](https://github.com/CitizensDev/Citizens2/commit/fe3f12e424b5c166821813ad91e51520d3cb9d9a) and [eb9764a](https://github.com/CitizensDev/Citizens2/commit/eb9764aeb07d8b53eb7273e7898504315361cd47)
